To ctor

kris-jusiak opened the pull request:
Problem:
-

Solution:
-

Issue: #

Reviewers:
@

kris-jusiak committed:
:new: `bind` to specific constructor

Problem:
- There is no simple way to bind type a specific constructor.

Solution:
- Add ability to bind type to a specific constructor via:
        * `bind<T>.to<T>(args...)`
        * `bind<T>(args...)`

Note:
- Specific constructor binding also fixes binding values in a specific
scope.
        * `bind<T>.to(value).in(scope)`

commit a53699ad337d4bff4e9db0d5b957d01b40ac240c
Author: Kris Jusiak <kris@jusiak.net>
Date:   Tue Oct 9 02:41:38 2018 +0000

    :bug: Fix #352 - Hello World program compilation fails with clang7

    Problem:
    - Clang7 has different behaviour when deducing templated implicit conversion operator using rvalue.

    Solution:
    - Add case for rvalue reference.

kris-jusiak committed:
:books: :fire: [doc] Remove fork ribbon

Problem:
- Fork ribbon is covering menu.

Solution:
- Remove it.

kris-jusiak committed:
:art: [extension] runtime_provider by default with shared_scope

Problem:
- Runtime provider extension is using singleton for deduced shared dependencies.

Solution:
- Rebind scope_traits to use the shared scope instead.

kris-jusiak committed:
:art: :new: [extension] runtime_provider - support for installation of runtime injector

Problem:
- Only compile time injectors can be installed.

Solution:
- Add support for installation of runtime injectors with install call.

kanstantsin-chernik commented on extension/include/boost/di/extension/providers/runtime_provider.hpp:
```diff
@@ -58,11 +60,21 @@ class runtime_provider : public config {
 
  public:
+  template <class T>
+  using scope_traits = di::extension::shared_config::scope_traits<T>;
```
It is better but I still a bit concerned about not being able to configure `runtime_provider`. May be we can have `runitme_provider` as a template to configure scope traits and policies?

kris-jusiak replied:
Done :+1:

kanstantsin-chernik commented on extension/include/boost/di/extension/providers/runtime_provider.hpp:
```diff
@@ -110,8 +127,15 @@ class injector : public core::injector<runtime_provider> {
-  template <class... Ts, class TInjector>
-  void install(aux::type_list<Ts...>, const TInjector &injector) {
+  template <class TInjector>
+  void install(aux::type_list<>, const TInjector &injector, aux::type<runtime_provider>) {
```
Just a reminder: it is not thread safe.

kris-jusiak replied:
Right :+1:

kris-jusiak committed:
:new: [extension] Injector

Problem:
- There is no extension to create easily create a differenty types of injectors.

Solution:
- Add extension::injector which allows to create different types of injector by only using injector type.
